cogs/verification.py

```diff
@@ -13,6 +15,43 @@
 from utils.utils import assign_student_roles, generateID
 
 
+@dataclass
+class BasicInfo:
+    branch: Optional[str]
+    section: Optional[str]
+    roll: Optional[int]
+
+
+def parse_verify_basic(params: str) -> BasicInfo:
+    roll = None
+    branch = None
+    section = None
+
+    if roll_no := re.search(r'\d{4,}', params):
+        roll = int(roll_no.group(0))
+
+    # try XY-A or XY-A2
+    if section_name := re.search(r'([A-Z]{2})[- ]?([ABC])', params, re.I):
+        branch = section_name.group(1)
+        section = section_name.group(2)
+
+    # try XY-01 or XY-1
+    elif sect := re.search(r'([A-Z]{2})[- ]?(\d{1,2})', params, re.I):
+        branch = sect.group(1)
+        sec_num = int(sect.group(2))
+        match sec_num:
+            case 1 | 2 | 3:
+                section = 'A'
+            case 4 | 5 | 6:
+                section = 'B'
+            case 7 | 8 | 9:
+                section = 'C'
+            case _:
+                section = None
```

Comment:
Revert back to basic if-else method as discord.py does not and will not support Python 3.10